addons/attributes/functions/fnc_gui_combo.sqf

```diff
@@ -20,8 +20,16 @@
  */
 
 params ["_controlsGroup", "_defaultValue", "_valueInfo"];
+
+//Value Info Eval
+if(typeName _valueInfo == typeName {}) then {
+    _valueInfo = call _valueInfo;
+    if(typeName _valueInfo != typeName []) then { _valueInfo = [] };
+};
+
 _valueInfo params [["_entries", [], [[]]]];
 
+
```

Comment:
I think adding `{}` to the allowed data types for `_entries` might be better. `_entries` is nested in `_valueInfo` to allow for other parameters to be easily added in the future. Something like:

```sqf
_valueInfo params [["_entries", [], [[], {}]]];

if (_entries isEqualType {}) then {
    _entries = call _entries;
};
```

Reply:
It would be more convenient to be able to dynamically generate all valueinfo subparameters than just entries, wouldn't it? Your proposal would limit the dynamic possibilities to just one parameter instead of all.

Reply:
Yes, at that point I would expect all control types to support `_valueInfo` as code. I guess the only one that collides with currently is `QGVAR(edit)` since it expects a sanitizing function which can be provided outside of an array. However, that is easily resolved by requiring that it is placed in an array.

Let's make all attribute controls support dynamic `_valueInfo` then.